File: src/Br/LotteryTicket/Lotteries/Welfare3D.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package Br.LotteryTicket.Lotteries;

import Br.LotteryTicket.Lottery;
import Br.LotteryTicket.LotteryTicket;
import Br.LotteryTicket.Result;
import Br.LotteryTicket.Ticket;
import Br.LotteryTicket.Utils;
import java.util.List;
import java.util.Random;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;

/**
 *
 * @author Bryan_lzh
 */
public class Welfare3D implements Lottery {

    private String Name = "福彩3D";
    private boolean Enable = false;
    private int Times;
    private double Price = 5d;
    private List<Result> Results;
    private Long Interval;
    private double v1;
    private double v2;
    private double v3;

    @Override
    public String getName() {
        return this.Name;
    }

    @Override
    public void setEnable(boolean b) {
        this.Enable = b;
    }

    @Override
    public boolean isEnable() {
        return this.Enable;
    }

    @Override
    public boolean isOK(int i) {
        if (i > 999) {
            return false;
        }
        if (i < 100) {
            return false;
        }
        return true;
    }

    @Override
    public long getInterval() {
        return this.Interval;// * 6L;
    }

    @Override
    public String Lottery() {
        Random random = new Random();
        int i = 0;
        while (true) {
            int var = random.nextInt(9);
            if (var != 0) {
                i += var * 100;
                break;
            }
        }
        i += random.nextInt(9) * 10;
        i += random.nextInt(9);
        this.Times++;
        this.Results.add(new Result(this.Name, this.Times, i));
        return "&e&l{福彩3D} 第[" + this.Times + "] 期彩票已经开奖 结果为: " + i + "  &b请有购买的拿起彩票右键领取";
    }

    @Override
    public double getPrice() {
        return this.Price;
    }

    @Override
    public void setTimes(int i) {
        this.Times = i;
    }

    @Override
    public int getTimes() {
        return this.Times;
    }

    @Override
    public void loadConfig(FileConfiguration config) {
        if (!config.contains("Lottery.Welfare3D.Price")) {
            if (config.contains("Lottery.Welfare3D.Pirce")) {
                config.set("Lottery.Welfare3D.Price", config.getDouble("Lottery.Welfare3D.Pirce"));
                config.set("Lottery.Welfare3D.Pirce", null);
            } else {
                config.set("Lottery.Welfare3D.Price", this.Price);
            }
        }
        if (!config.contains("Lottery.Welfare3D.Interval_单位小时")) {
            config.set("Lottery.Welfare3D.Interval_单位小时", 6.0);
        }
        if (!config.contains("Lottery.Welfare3D.中1个数字的奖励")) {
            config.set("Lottery.Welfare3D.中1个数字的奖励", 5d);
            config.set("Lottery.Welfare3D.中2个数字的奖励", 10d);
            config.set("Lottery.Welfare3D.中3个数字的奖励", 20d);
        }
        this.Price = config.getDouble("Lottery.Welfare3D.Price");
        this.Interval = Long.valueOf((20 * 60 * 60 * config.getDouble("Lottery.Welfare3D.Interval_单位小时") + "").split("\\.")[0]).longValue();
        this.v1 = config.getDouble("Lottery.Welfare3D.中1个数字的奖励");
        this.v2 = config.getDouble("Lottery.Welfare3D.中2个数字的奖励");
        this.v3 = config.getDouble("Lottery.Welfare3D.中3个数字的奖励");
    }

    @Override
    public List<Result> getResults() {
        return this.Results;
    }

    @Override
    public void setResults(List<Result> list) {
        this.Results = list;
    }

    @Override
    public void Award(Player p, Ticket ticket, Result r) {
        int t[] = Utils.toIntArray(ticket.Number);
        int re[] = Utils.toIntArray(r.getNumber());
        int i = 0;
        int a = 0;
        for (int e : re) {
            if (t[i] == e) {
                a++;
            }
            i++;
            if (i > 2) {
                break;
            }
        }
        switch (a) {
            case 0:
                p.sendMessage(Utils.sendMessage("&6&l可惜 你没有获得任何奖励QAQ"));
                break;
            case 1:
                LotteryTicket.econ.depositPlayer(p, this.v1 * ticket.Amount);
                p.sendMessage(Utils.sendMessage("&e&l恭喜 你中了一个数字 获得: " + this.v1 * ticket.Amount + LotteryTicket.econ.currencyNamePlural()));
                break;
            case 2:
                LotteryTicket.econ.depositPlayer(p, this.v2 * ticket.Amount);
                p.sendMessage(Utils.sendMessage("&e&l恭喜 你中了二个数字 获得: " + this.v2 * ticket.Amount + LotteryTicket.econ.currencyNamePlural()));
                break;
            case 3:
                LotteryTicket.econ.depositPlayer(p, this.v3 * ticket.Amount);
                p.sendMessage(Utils.sendMessage("&e&l恭喜 你中了三个数字 获得: " + this.v3 * ticket.Amount + LotteryTicket.econ.currencyNamePlural()));
                break;
        }
    }

    @Override
    public String getCode() {
        return "Welfare3D";
    }

    @Override
    public String[] getCommandName() {
        return new String[]{
                    "福彩3D",
                    "3D",
                    "福利彩票3D",
                    "Welfare3D"
                };
    }

    @Override
    public boolean isOK(String s) {
        return false;
    }

    @Override
    public Type getType() {
        return Type.Int;
    }

    @Override
    public String[] getUsage() {
        return new String[]{
            "§6[选购的数字]: 连续的&l3&6个数字 注意 首相不能为0",
            "^b例如 : 156 | 559"
        };
    }
}

File: src/Welfare3D.js
// @Name 福彩3D
// @Usage §6[选购的数字]: 连续的§l3§6个数字 注意 首项不能为0
// @Usage §b例如 : 156 | 559
// @Code Welfare3D
// @Default Price 5
// @Default Interval 1天
// @Default Enable true
// @Type Int
// @Command 福彩3D,3D,福利彩票3D,Welfare3D
var Random = new java.util.Random();
var Utils = Java.type('Br.LotteryTicket.Utils');
var LastResult;
/**
 * 初始化函数
 * @returns {undefined} 无返回值
 */
function init() {
}
/**
 * 判断用户输入的参数是否合法
 * @param {int|String} input 用户输入的参数
 * @returns {Boolean} 合法时返回true
 */
function isValid(input) {
    if (input > 999 || input < 100) {
        return false;
    }
    return true;
}

/**
 * 开奖时调用
 * 其中返回值里 支持颜色转意
 * %times%表示期数
 * %result%表示结果
 * @returns {String} 返回开奖结果(需要广播的)
 */
function Lottery() {
    var i = 0;
    while (i < 99) {
        var v = Random.nextInt(9);
        i = i * 10 + v;
    }
    LastResult = i;
    return '&e&l{福彩3D} 第[%times%] 期彩票已经开奖 结果为: %result% &b请有购买的拿起彩票右键领取';
}
/**
 * 返回最后一次开奖的结果
 * @returns int|String 可以返回两者之一
 */
function getLastResult() {
    return LastResult;
}
var WinAward = [5.0, 10.0, 20.0];
/**
 * 读取配置 同时也可以在此写入配置
 * @param {FileConfiguration} config 
 * @returns {undefined} 无返回值
 */
function loadConfig(config) {
    if (!config.contains('WinAward.1')) {
        config.set('WinAward.1', WinAward[0]);
        config.set('WinAward.2', WinAward[1]);
        config.set('WinAward.3', WinAward[2]);
    } else {
        WinAward[0] = config.getDouble('WinAward.1');
        WinAward[1] = config.getDouble('WinAward.2');
        WinAward[2] = config.getDouble('WinAward.3');
    }
}

/**
 * 奖励玩家的方式
 * @param {Player} p 玩家
 * @param {Ticker} ticket 玩家所使用的票
 * @param {Result} result 开奖结果
 * @returns {undefined} 无返回值
 */
function onAward(p, ticket, result) {
    var t = Utils.toIntArray(ticket.getNumber());
    var re = Utils.toIntArray(r.getNumber());
    var a = 0;
    for (var index = 0; index < re.length; index++) {
        var e = re[index];
        if (e === t[index]) {
            a++;
        }
        if (index > 2) {
            break;
        }
    }
    switch (a) {
        case 0:
            p.sendMessage(Utils.sendMessage("&6&l可惜 你没有获得任何奖励QAQ"));
            break;
        case 1:
            Utils.payToPlayer(p, WinAward[0] * ticket.getAmount());
            p.sendMessage(Utils.sendMessage("&e&l恭喜 你中了一个数字 获得: " + WinAward[0] * ticket.getAmount()));
            break;
        case 2:
            Utils.payToPlayer(p, WinAward[1] * ticket.getAmount());
            p.sendMessage(Utils.sendMessage("&e&l恭喜 你中了二个数字 获得: " + WinAward[1] * ticket.getAmount()));
            break;
        case 3:
            Utils.payToPlayer(p, WinAward[2] * ticket.getAmount());
            p.sendMessage(Utils.sendMessage("&e&l恭喜 你中了三个数字 获得: " + WinAward[2] * ticket.getAmount()));
            break;
    }
}
